components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/AbstractComplexController.java

```diff
@@ -127,7 +127,9 @@ public void writeJSON(Patient patient, JSONObject json, Collection<String> selec
             Map.Entry<String, T> item = iterator.next();
             String itemKey = item.getKey();
             Object formattedValue = format(itemKey, item.getValue());
-            if (formattedValue != null && (selectedFieldNames == null || selectedFieldNames.contains(item.getKey()))) {
+            if (formattedValue != null
+                && (selectedFieldNames == null
+                || selectedFieldNames.contains(getControllingFieldName(item.getKey())))) {
```

Comment:
I would put this on the previous line since || is not on the same level as &&, but is not worth fixing if that is the only change, just a comment